src/coreclr/tools/superpmi/superpmi-shared/spmiutil.cpp

```diff
@@ -99,86 +101,84 @@ WCHAR* GetEnvironmentVariableWithDefaultW(const WCHAR* envVarName, const WCHAR*
     return retString;
 }
 
-#ifdef TARGET_UNIX
-// For some reason, the PAL doesn't have GetCommandLineA(). So write it.
-LPSTR GetCommandLineA()
+const char* GetEnvWithDefault(const char* envVarName, const char* defaultValue)
 {
-    LPSTR  pCmdLine  = nullptr;
-    LPWSTR pwCmdLine = GetCommandLineW();
-
-    if (pwCmdLine != nullptr)
-    {
-        // Convert to ASCII
-
-        int n = WideCharToMultiByte(CP_ACP, 0, pwCmdLine, -1, nullptr, 0, nullptr, nullptr);
-        if (n == 0)
-        {
-            LogError("MultiByteToWideChar failed %d", GetLastError());
-            return nullptr;
-        }
+    // getenv isn't thread safe, but it's simple and sufficient since we are development-only tool
+    char* env = getenv(envVarName);
+    return env ? env : defaultValue;
+}
 
-        pCmdLine = new char[n];
+std::string GetProcessCommandLine()
+{
+#ifdef TARGET_WINDOWS
+    return ::GetCommandLineA();
+#else
+    char cmdLine[256];
```

Comment:
This often isn't going to be sufficient. I would use `strlen` to compute length, allocate, and then free and converting to `std::string`.

Reply:
The command line is used to construct output file name, with a maximum length. I'll check the usage.